### ISSUE TITLE
MicrosoftAjax.js fails with 'Access is Denied" message on wcf service call over https

### ISSUE THREAD

**Issue by danishtaqvi** [DevExpress/AjaxControlToolkit#101]
Our application was running with AjaxControlToolkit v9.11 over .net2.0 and making calls to wcf web service over http and https and was working properly. application was referring to external script files under scriptmanager (previously ToolkitScriptManager) as below

```
<asp:scriptmanager id="sm" scriptmode="Release" enablepartialrendering="true" runat="server">
          <Scripts>
            <asp:ScriptReference Name="MicrosoftAjax.js" Path="scripts/MicrosoftAjax/MicrosoftAjax.js" />
            <asp:ScriptReference Name="MicrosoftAjaxWebForms.js" Path="scripts/MicrosoftAjax/MicrosoftAjaxWebForms.js" />
        </Scripts>
    <services>
            <asp:servicereference path="~/PrintService.svc" />
        </services>
</asp:scriptmanager>
```

  We upgraded our application to .net4.5.2 and since then wcf web service call over secure https is giving the 'Access is denied' error. It works fine over http. 
  We upgraded AjaxControlToolkit version to 16.1 using Nuget package, however, external script files under "/scripts/MicrosoftAjax/" folder are not updated and so MicrosoftAjax.js file is still the same and we still get the same error over https. 
  How do I resolve this error? Where can I get the external script files or does the new AjaxControlToolkit version doesn't need to have external script file references?

error from fiddler:
stack
```
"Error: Access is denied.
   at a.Net.XMLHttpExecutor.prototype.executeRequest (https://securewebtest.vfsgroup.com/TransActionEOS/scripts/MicrosoftAjax/MicrosoftAjax.js:1:80534)
   at a.Net._WebRequestManager.prototype.executeRequest (https://securewebtest.vfsgroup.com/TransActionEOS/scripts/MicrosoftAjax/MicrosoftAjax.js:1:83266)
   at a.Net.WebRequest.prototype.invoke (https://securewebtest.vfsgroup.com/TransActionEOS/scripts/MicrosoftAjax/MicrosoftAjax.js:1:85065)
   at a.Net.WebServiceProxy.invoke (https://securewebtest.vfsgroup.com/TransActionEOS/scripts/MicrosoftAjax/MicrosoftAjax.js:1:88588)
   at a.Net.WebServiceProxy.prototype._invoke (https://securewebtest.vfsgroup.com/TransActionEOS/scripts/MicrosoftAjax/MicrosoftAjax.js:1:87236)
   at PrintService.prototype.PrintDocList (https://securewebtest.vfsgroup.com/TransActionEOS/PrintService.svc/js:14:1)
   at PrintService.PrintDocList (https://securewebtest.vfsgroup.com/TransActionEOS/PrintService.svc/js:38:131)
   at GetDocs (https://securewebtest.vfsgroup.com/TransActionEOS/doclist.aspx:369:21)
   at Global code (https://securewebtest.vfsgroup.com/TransActionEOS/doclist.aspx:468:833)"
```

**Comment by AlekseyMartynov** [comment]
@danishtaqvi
Those scripts, MicrosoftAjax.js and MicrosoftAjaxWebForms.js, are a part of the [Microsoft.AspNet.ScriptManager.MSAjax](https://www.nuget.org/packages/Microsoft.AspNet.ScriptManager.MSAjax/) NuGet package. Add it to the project, or update if it's been installed earlier. Note that the new scripts are placed to a slightly different location: ~/scripts/WebForms/MSAjax

**Comment by danishtaqvi** [comment]
Thanks @AlekseyMartynov for the quick response and pointing me to the package.

Ok, so I added the package Microsoft.AspNet.ScriptManager.MSAjax to the project and got the scripts folder and files created; updated the reference in my page but got the same error from the new MicrosoftAjax.js file this time. Error in fiddler stack:

"Error: Access is denied.\r\n\n   at Sys.Net.XMLHttpExecutor.prototype.executeRequest (https://securewebtest.vfsgroup.com/TransActionEOS/scripts/WebForms/MSAjax/MicrosoftAjax.js:6:85275)\n   at Sys.Net._WebRequestManager.prototype.executeRequest (https://securewebtest.vfsgroup.com/TransActionEOS/scripts/WebForms/MSAjax/MicrosoftAjax.js:6:88634)\n   at Sys.Net.WebRequest.prototype.invoke (https://securewebtest.vfsgroup.com/TransActionEOS/scripts/WebForms/MSAjax/MicrosoftAjax.js:6:90645)\n   at Sys.Net.WebServiceProxy.invoke (https://securewebtest.vfsgroup.com/TransActionEOS/scripts/WebForms/MSAjax/MicrosoftAjax.js:6:97335)\n   at Sys.Net.WebServiceProxy.prototype._invoke (https://securewebtest.vfsgroup.com/TransActionEOS/scripts/WebForms/MSAjax/MicrosoftAjax.js:6:95743)\n   at PrintService.prototype.PrintDocList (https://securewebtest.vfsgroup.com/TransActionEOS/PrintService.svc/js:14:1)\n   at PrintService.PrintDocList (https://securewebtest.vfsgroup.com/TransActionEOS/PrintService.svc/js:38:131)\n   at GetDocs (https://securewebtest.vfsgroup.com/TransActionEOS/doclist.aspx:371:21)\n   at Global code (https://securewebtest.vfsgroup.com/TransActionEOS/doclist.aspx:470:871)"

Since this was working when the project was on .net2.0, and upgrading it to .net4.5.2 had made changes to web.config file, I want to highlight that some sections of web.config have been removed with the upgrade which looks important; 

`    <configSections>
        <sectionGroup name="system.web.extensions" type="System.Web.Configuration.SystemWebExtensionsSectionGroup, System.Web.Extensions, Version=3.5.0.0, Culture=neutral, PublicKeyToken=31BF3856AD364E35">
            <sectionGroup name="scripting" type="System.Web.Configuration.ScriptingSectionGroup, System.Web.Extensions, Version=3.5.0.0, Culture=neutral, PublicKeyToken=31BF3856AD364E35">
                <section name="scriptResourceHandler" type="System.Web.Configuration.ScriptingScriptResourceHandlerSection, System.Web.Extensions, Version=3.5.0.0, Culture=neutral, PublicKeyToken=31BF3856AD364E35" requirePermission="false" allowDefinition="MachineToApplication" />
                <sectionGroup name="webServices" type="System.Web.Configuration.ScriptingWebServicesSectionGroup, System.Web.Extensions, Version=3.5.0.0, Culture=neutral, PublicKeyToken=31BF3856AD364E35">
                    <section name="jsonSerialization" type="System.Web.Configuration.ScriptingJsonSerializationSection, System.Web.Extensions, Version=3.5.0.0, Culture=neutral, PublicKeyToken=31BF3856AD364E35" requirePermission="false" allowDefinition="Everywhere" />
                    <section name="profileService" type="System.Web.Configuration.ScriptingProfileServiceSection, System.Web.Extensions, Version=3.5.0.0, Culture=neutral, PublicKeyToken=31BF3856AD364E35" requirePermission="false" allowDefinition="MachineToApplication" />
                    <section name="authenticationService" type="System.Web.Configuration.ScriptingAuthenticationServiceSection, System.Web.Extensions, Version=3.5.0.0, Culture=neutral, PublicKeyToken=31BF3856AD364E35" requirePermission="false" allowDefinition="MachineToApplication" />
                    <section name="roleService" type="System.Web.Configuration.ScriptingRoleServiceSection, System.Web.Extensions, Version=3.5.0.0, Culture=neutral, PublicKeyToken=31BF3856AD364E35" requirePermission="false" allowDefinition="MachineToApplication" />
                </sectionGroup>
            </sectionGroup>
        </sectionGroup>
    </configSections>
    <system.web>
        <pages>
            <controls>
                <add tagPrefix="asp" namespace="System.Web.UI" assembly="System.Web.Extensions, Version=3.5.0.0, Culture=neutral, PublicKeyToken=31BF3856AD364E35" />
                <add tagPrefix="asp" namespace="System.Web.UI.WebControls" assembly="System.Web.Extensions, Version=3.5.0.0, Culture=neutral, PublicKeyToken=31BF3856AD364E35" />
            </controls>`

and

`       <compilation debug="true" strict="false" explicit="true">
            <assemblies>
                <add assembly="Microsoft.VisualBasic.Compatibility, Version=8.0.0.0, Culture=neutral, PublicKeyToken=B03F5F7F11D50A3A" />
                <add assembly="System.Design, Version=2.0.0.0, Culture=neutral, PublicKeyToken=B03F5F7F11D50A3A" />
               <add assembly="System.Security, Version=2.0.0.0, Culture=neutral, PublicKeyToken=B03F5F7F11D50A3A" />
               <add assembly="System.Web.Extensions, Version=3.5.0.0, Culture=neutral, PublicKeyToken=31BF3856AD364E35" />`

and runtime section referring to System.Web.Extensions. I have attached both config files (2.0 and 4.5.2) to show the difference.

[web.config.zip](https://github.com/DevExpress/AjaxControlToolkit/files/259999/web.config.zip)

**Comment by AlekseyMartynov** [comment]
Changes to web.config seem fine, and it's unlikely that they are related to the "Access denied" error. It rather looks like a cross-domain issue, and may be caused by the fact that WCF Services assemblies have been updated too.

Here are a couple of StackOverflow threads discussing the same problem:

* http://stackoverflow.com/q/25889599
* http://stackoverflow.com/q/15529328

**Comment by danishtaqvi** [comment]
Thanks for sharing those @AlekseyMartynov . 
We did actually visited the cross-domain issue articles earlier and rules out the possibility of this cause as the service is hosted within the same application/domain. Since this error comes in only with .net4.5.2 framework and not with .net2.0, it seem to be internal to working of .net and AjaxControlToolkit. 
Eventually, I ran out of options to make it work with AjaxControlToolkit, so tried jquery ajax call and it worked. There was nothing much I had to change for that and below is the working code snippet:

`                    $.ajax({
                        type: 'POST',
                        contentType: "application/json; charset=utf-8",
                        url: 'PrintService.svc/PrintDocList',
                        data: JSON.stringify({parameter1: value1, parameter2: value2}),
                        dataType: "json",
                        success: GetDocsSuccess,
                        error: GetDocsFailed
                    });`

Thanks for your kind support, really appreciate your help in this regard.
